src/Utopia/Messaging/Adapter/Push/FCM.php

```diff
@@ -144,10 +144,20 @@ protected function process(PushMessage $message): array
             if ($result['statusCode'] === 200) {
                 $response->incrementDeliveredTo();
             }
-            $response->addResultForRecipient(
-                $message->getTo()[$index],
-                $result['response']['error']['message'] ?? ''
-            );
+
+            if (isset($result['response']['error'])) {
+                $response->addResultForRecipient(
+                    $message->getTo()[$index],
+                    $result['response']['error']['status'] === 'UNREGISTERED' ||
+                    $result['response']['error']['status'] === 'INVALID_ARGUMENT'
+                    ? 'Expired device token.'
+                    : $result['response']['error']['message'] ?? ''
```

Comment:
Why are we doing this and dropping the error message from the provider?

Reply:
Each push type provider has a different exception message for when tokens expire. For push tokens, we are specifying an error message for expired device token. We decided to delete push targets which receive invalid token or expired token error. This error message has actually moved to Push Adapter base class. 

cc: @abnegate